nvtabular/dispatch.py

```diff
@@ -13,14 +13,74 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+import itertools
 from typing import Union
 
 import cudf
+import cupy as cp
+import numpy as np
 import pandas as pd
+import pyarrow.parquet as pq
+from cudf.core.column import as_column, build_column
+from cudf.utils.dtypes import is_list_dtype
+from dask.dataframe.utils import hash_object_dispatch
 
 DataFrameType = Union[pd.DataFrame, cudf.DataFrame]
 
 
+def _arange(size, like_df=None):
+    """Dispatch for numpy.arange"""
+    if isinstance(like_df, pd.DataFrame):
+        return np.arange(size)
+    else:
+        return cp.arange(size)
+
+
+def _hash_series(s):
+    """Row-wise Series hash"""
+    if isinstance(s, pd.Series):
+        return hash_object_dispatch(s).values
+    else:
+        if _is_list_dtype(s):
+            return s.list.leaves.hash_values()
+        else:
+            return s.hash_values()
```

Comment:
I might be super paranoid here - but the gpu/cpu versions here don't return the same results (different hash functions) and I'm wondering if this will cause issues down the line.

Like if we transform a dataset during training on the gpu - and then try to do the same transforms on the cpu at inference time - and the hash operations will be different so everything will break.

Should we put a warning on the CPU version for now? Ideally we could also do the same hashing operations on the cpu too (cudf does murmurhash3 iirc)

Reply:
Ah - I was actually thinking the same thing, but forgot to leave a note.  I'll think about this a bit and report back here.

Reply:
Okay - I added a note and a warning for now.  I also tried playing with `mmh3`, but I'm having trouble reproducing cudf hash.

```python
import cudf
import numpy as np
import mmh3

def _hash(val):
    return mmh3.hash_from_buffer(np.array(val))
func = np.frompyfunc(_hash, 1, 1)

print("mmh3:", func(np.arange(2)))
print("cudf:", cudf.Series(np.arange(2)).hash_values())
```
gives...
```
mmh3: [1669671676 1392991556] 
cudf: [  29140149 -247539971] 
```

However, this is likely operator error, since I'm not really sure how to use `mmh3` "correctly"

Reply:
I've created an issue to track https://github.com/NVIDIA/NVTabular/issues/569 - I think solving this is out of scope of this PR